src/nbl/video/IGPUDescriptorSet.cpp

```diff
@@ -201,18 +223,24 @@ void IGPUDescriptorSet::processCopy(const IGPUDescriptorSet::SCopyDescriptorSet&
         auto* dstDescriptors = copy.dstSet->getDescriptors(type, copy.dstBinding);
         assert(!(!srcDescriptors != !dstDescriptors));
 
-        auto* srcSamplers = copy.srcSet->getMutableSamplers(copy.srcBinding);
-        auto* dstSamplers = copy.dstSet->getMutableSamplers(copy.dstBinding);
+        auto* srcSamplers = copy.srcSet->getMutableCombinedSamplers(copy.srcBinding);
+        auto* dstSamplers = copy.dstSet->getMutableCombinedSamplers(copy.dstBinding);
         assert(!(!srcSamplers != !dstSamplers));
 
         if (srcDescriptors && dstDescriptors)
             std::copy_n(srcDescriptors, copy.count, dstDescriptors);
 
         if (srcSamplers && dstSamplers)
             std::copy_n(srcSamplers, copy.count, dstSamplers);
+
+        auto& bindingRedirect = m_layout->getDescriptorRedirect(type);
+        auto bindingCreateFlags = bindingRedirect.getCreateFlags(bindingRedirect.findBindingStorageIndex({ copy.srcBinding }));
```

Comment:
again amortize the lookup of the storage redirect from the binding, like here https://github.com/Devsh-Graphics-Programming/Nabla/pull/693/files#r1649030085